### PR TITLE
Skip Task Profile/Traceback unit tests

### PR DESCRIPTION
**Pull request by chaowanggg**
We will skip unit tests for the task profile/traceback because the mac and win32 systems do not support the execution of the `py-spy` command, which is used in generating task profiles and tracebacks.

## Why is this critical for Ray 2.7?
We plan to launch this feature in Ray 2.7. However, the failure of the mac test cases is a blocker for its release.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/reporter/tests/test_reporter.py [hunk 1]
```diff
@@ -696,13 +696,15 @@ def _generate_worker_key(self, proc):
...
-
-
+@pytest.mark.skipif(
+    os.environ.get("RAY_MINIMAL") == "1",
+    reason="This test is not supposed to work for minimal installation.",
+)
+@pytest.mark.skipif(sys.platform == "win32", reason="No py-spy on Windows.")
+@pytest.mark.skipif(
+    sys.platform == "darwin",
+    reason="Fails on OSX: https://github.com/ray-project/ray/issues/30114",
```
Comment (chaowanggg) [inline dashboard/modules/reporter/tests/test_reporter.py:706]: I linked the pr since it will provide more context about py-spy

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
can you pick this to. ray 2.7?

**Comment by chaowanggg** [comment]
> 

Sure, I will do this

**Comment by zhe-thoughts** [comment]
@chaowanggg Thanks for the work. Let's fill in necessary info on the PR, especially on why this is critical for Ray 2.7. Thanks!

**Comment by chaowanggg** [comment]
> @chaowanggg Thanks for the work. Let's fill in necessary info on the PR, especially on why this is critical for Ray 2.7. Thanks!

Thanks for the remind. Done!

**Pull request merged**